arangod/RocksDBEngine/RocksDBKeyBounds.cpp

```diff
@@ -427,9 +427,18 @@ RocksDBKeyBounds::RocksDBKeyBounds(RocksDBEntryType type, uint64_t first,
         uint64ToPersistent(_internals.buffer(), first);
         _internals.buffer().append((char const*)(max.begin()), max.byteSize());
       } else {
-        // in case of forward iteration, we can use the next objectId as a quick
-        // termination case, as it will be in the next prefix
-        uint64ToPersistent(_internals.buffer(), first + 1);
+        // in case of forward iteration, we can use the next objectId as
+        // a quick termination case, as it will be in the next prefix,
+        // however, this trick only works when we have the big endian
+        // data format:
+        if (rocksDBEndianness == RocksDBEndianness::Big) {
+          uint64ToPersistent(_internals.buffer(), first + 1);
+        } else {
+          // little endian:
```

Comment:
We could add an assert here, in case there will be a third type some when in the future,
e.g. "TRI_ASSERT(rocksutils::rocksDBEndianness == RocksDBEndianness::Little);"